experimental/sdk/webpubsub-socketio-extension/src/SIO/index.ts

```diff
@@ -1,25 +1,29 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { WebPubSubExtensionOptions } from "../common/utils";
+import { debugModule, WebPubSubExtensionOptions } from "../common/utils";
 import { WebPubSubEioServer } from "../EIO";
 import { WebPubSubAdapterProxy } from "./components/web-pubsub-adapter";
 import * as SIO from "socket.io";
 
+const debug = debugModule("wps-sio-ext:SIO:index");
+debug("load");
+
 export function useAzureWebPubSub(
   this: SIO.Server,
   webPubSubOptions: WebPubSubExtensionOptions,
   useDefaultAdapter: boolean = true
 ): SIO.Server {
-  const engine = new WebPubSubEioServer(this.engine.opts, webPubSubOptions);
+  debug("use Azure Web PubSub For Socket.IO Server");
 
+  const engine = new WebPubSubEioServer(this.engine.opts, webPubSubOptions);
   engine.attach((this as any).httpServer, (this as any).opts);
-
   this.bind(engine as any);
 
   if (!useDefaultAdapter) {
-    const adapterProxy = new WebPubSubAdapterProxy("NotImplementedArg");
+    debug("use webPubSub adatper");
 
+    const adapterProxy = new WebPubSubAdapterProxy((this.engine as any).webPubSubConnectionManager.serviceClient)
```

Comment:
why `as any`? can it be strong typed?

Reply:
it could. And it will lead compile failure, for this PR doesn't have completed class implementation of `this.engine`. I updated